app/controllers/topics_controller.rb

```diff
@@ -1396,17 +1396,18 @@ def bulk_unread_topic_ids
           topic_query.joined_topic_user,
           whisperer: guardian.is_whisperer?,
         ).listable_topics
+
       topics = TopicQuery.tracked_filter(topics, current_user.id) if params[:tracked].to_s == "true"
 
       if params[:category_id]
-        if params[:include_subcategories]
-          topics = topics.where(<<~SQL, category_id: params[:category_id])
-            category_id in (select id FROM categories WHERE parent_category_id = :category_id) OR
-            category_id = :category_id
-          SQL
-        else
-          topics = topics.where("category_id = ?", params[:category_id])
-        end
+        category_ids =
+          if params[:include_subcategories]
+            Category.subcategory_ids(params[:category_id].to_i)
```

Comment:
Uh just to confirm `Category.subcategory_ids(params[:category_id].to_i)` returns sub sub categories for a category?

Reply:
Yup it returns an array which contains the category id passed to the method as well as the ids of the sub-category and sub-sub-categories for the given category id.